### PR TITLE
Removing grid lowering

### PR DESCRIPTION
**Pull request by apbose**
This PR addresses the issue #2665 
1. Removing grid_sampler_2d from lowering ops
2. Adding grid_sampler.default and grid_sampler_2d.default to the CONVERTER key registry
3. Adding tests for grid_sampler.default and grid_sampler_2d.default tests
4. Restructing the test
5. Correcting the interpolation mode and padding mode in the tests

### CODE REVIEW

**Comment by apbose** [comment]
PR #2682 seems to adddress partially the above.

File: tests/py/dynamo/conversion/test_grid_aten.py [hunk 1]
```diff
@@ -126,19 +88,98 @@ class TestGridConverter(DispatchTestCase):
...
+    )
+    def test_grid(self, _, op_name, op, input_shape, dim_shape):
+        class TestModule(nn.Module):
+            def __init__(self, grid_sampler_op):
+                super().__init__()
+                self.grid_sampler_op = grid_sampler_op
+
+            def forward(self, x):
+                grid = torch.randint(-1, 1, dim_shape, dtype=torch.float32)
+                return self.grid_sampler_op(x, grid, grid_sampler_aten_ops[op_name])
```
Comment (gs-olive) [inline tests/py/dynamo/conversion/test_grid_aten.py:104]: Would this need to be `grid_sampler_aten_ops[op_name](x, grid)`?

Comment (apbose) [reply tests/py/dynamo/conversion/test_grid_aten.py:104]: @gs-olive yes it could be but then in that case it would be something like-
```
   def test_grid(self, _, op_name, input_shape, dim_shape, padding_mode, interpolation_mode, align_corners):
        class TestModule(nn.Module):
            def __init__(self):
                super().__init__()

            def forward(self, x):
                grid = torch.randint(-1, 1, dim_shape, dtype=torch.float32)
                return grid_sampler_aten_ops[op_name](x, grid, padding_mode, interpolation_mode, align_corners)

        inputs = [torch.randn(input_shape, dtype=torch.float32)]
        grid_model = TestModule()
        self.run_test(grid_model, inputs)
```
It would be code design choice to either  encapsulate it in the lambda function in the parameters, or declare it in the above way. In my opinion both should be good.

You could let me know if you think otherwise.

Comment (gs-olive) [reply tests/py/dynamo/conversion/test_grid_aten.py:104]: I see - this makes sense, thanks. In that case, it seems good to go - should just need a rebase to enable testing.
